avalon/maya/commands.py

```diff
@@ -33,17 +34,29 @@ def reset_frame_range():
         cmds.warning("No edit information found for %s" % shot["name"])
         return
 
-    fps = {
-        "12": "12fps",
-        "15": "game",
-        "16": "16fps",
-        "24": "film",
-        "25": "pal",
-        "30": "ntsc",
-        "48": "show",
-        "50": "palf",
-        "60": "ntscf"
-    }.get(api.Session.get("AVALON_FPS"), "pal")  # Default to "pal"
+    # this handles fps set as decimal number. For example
+    # fps 25.0 is corrected to int so mapping will still work
+    avalon_fps = api.Session.get("AVALON_FPS")
+    d, i = math.modf(float(api.Session.get("AVALON_FPS", 25)))
+    if d == 0.0:
+        avalon_fps = int(i)
+
+    fps = {'15': 'game',
+           '24': 'film',
+           '25': 'pal',
+           '30': 'ntsc',
+           '48': 'show',
+           '50': 'palf',
+           '60': 'ntscf',
+           '23.98': '23.976fps',
+           '23.976': '23.976fps',
+           '29.97': '29.97fps',
+           '47.952': '47.952fps',
+           '47.95': '47.952fps',
+           '59.94': '59.94fps',
+           '44100': '44100fps',
+           '48000': '48000fps'
+           }.get(str(avalon_fps), "pal")  # Default to "pal"
```

Comment:
I think we could just use float or int as key instead of strings, that way, you won't need `math.modf` to ensure the string representation of FPS value.

For example:

```python
FPS_MAP = {
    24: "film",
}
assert FPS_MAP[24] == "film"
assert FPS_MAP[24.0] == "film"
```

So we could just convert the value of "AVALON_FPS" into float and directly map to the Maya FPS.

Reply:
The danger would I guess be a potential floating point precision problem? But if that doesn't happen then maybe yes, this could be simplified. @antirotor could you try David's approach out for your cases and see if it results in issues?

Reply:
This is good approach, more clean I would say. But it would require changes in config or other places where when using AVALON_FPS we would need to convert from str to float, as environment variable should have its value in str anyway. So I think we can either go with cleaner solution breaking dependent code slightly or hack it like I did but probably without need to change anything else.

Reply:
> But it would require changes in config or other places where when using AVALON_FPS we would need to convert from str to float, as environment variable should have its value in str anyway.

I think neither of us would need to change anything in any config since the *str-to-float* conversion already happened [here] in this PR, right ? :thinking:

Reply:
You are absolutely right. Fixed it.